helix-term/src/commands/lsp.rs

```diff
@@ -64,28 +64,78 @@ fn jump_to_location(
     align_view(doc, view, Align::Center);
 }
 
+/// Returns a string showing the symbol nested in its containing symbol if it exists.
+///
+/// # Arguments
+/// * `symbol` - the symbol that we're creating the string for
+///
+/// # Example
+/// If you have something like:
+/// ```
+/// mod foo {
+///     fn bar() {}
+/// }
+/// ```
+/// the result will be: `function  : foo → bar`
+fn get_symbol_string(symbol: &lsp::SymbolInformation) -> String {
+    let kind = match symbol.kind {
+        lsp::SymbolKind::FILE => "file      ",
+        lsp::SymbolKind::MODULE => "module    ",
+        lsp::SymbolKind::NAMESPACE => "namespace ",
+        lsp::SymbolKind::PACKAGE => "package   ",
+        lsp::SymbolKind::CLASS => "class     ",
+        lsp::SymbolKind::METHOD => "method    ",
+        lsp::SymbolKind::PROPERTY => "property  ",
+        lsp::SymbolKind::FIELD => "field     ",
+        lsp::SymbolKind::CONSTRUCTOR => "construct ",
+        lsp::SymbolKind::ENUM => "enum      ",
+        lsp::SymbolKind::INTERFACE => "interface ",
+        lsp::SymbolKind::FUNCTION => "function  ",
+        lsp::SymbolKind::VARIABLE => "variable  ",
+        lsp::SymbolKind::CONSTANT => "constant  ",
+        lsp::SymbolKind::STRING => "string    ",
+        lsp::SymbolKind::NUMBER => "number    ",
+        lsp::SymbolKind::BOOLEAN => "boolean   ",
+        lsp::SymbolKind::ARRAY => "array     ",
+        lsp::SymbolKind::OBJECT => "object    ",
+        lsp::SymbolKind::KEY => "key       ",
+        lsp::SymbolKind::NULL => "null      ",
+        lsp::SymbolKind::ENUM_MEMBER => "enum mem  ",
+        lsp::SymbolKind::STRUCT => "struct    ",
+        lsp::SymbolKind::EVENT => "event     ",
+        lsp::SymbolKind::OPERATOR => "operator  ",
+        lsp::SymbolKind::TYPE_PARAMETER => "type param",
+        _ => "",
+    };
```

Comment:
I don't see much benefits of aligning this, since they have a different group.

Reply:
The benefit is that it makes it easier for eyes to spot the symbol kind and the symbol name. The screenshot below shows the difference. The top (with alignment) makes it much easier to see at a glance. Is there any specific disadvantage that you are thinking of?

![Screen Shot 2022-05-14 at 7 06 42 PM](https://user-images.githubusercontent.com/665342/168454209-e5675fdd-fac1-4027-80c2-477ac62766ad.png)

Reply:
Wouldn't that be solved if we just reorder the appearance? Or maybe the order is somehow useful?